tests/tests/test_mender_connect.py

```diff
@@ -180,3 +181,70 @@ def test_bogus_proto_message(
             assert prot.props["status"] == protomsg.PROP_STATUS_ERROR
             assert prot.protoType == 12345
             assert prot.typ == proto_shell.MSG_TYPE_SPAWN_SHELL
+
+    def test_session_recording(
+        self, class_persistent_standard_setup_one_client_bootstrapped
+    ):
+        def get_cmd(ws, timeout=1):
+            pmsg = protomsg.ProtoMsg(1)
+            body = b""
+            while True:
```

Comment:
The outer `while` is redundant, no? The inner `while` has only one exit condition, the exception, which immediately returns.